src/main.rs

```diff
@@ -295,13 +295,48 @@ async fn process_feed(
             .permalink(false)
             .build();
 
-        let rss_item = ItemBuilder::default()
+        let mut rss_item_builder = ItemBuilder::default();
+        rss_item_builder
             .title(title_text)
             .link(base_url.parse(link_url).ok().map(|u| u.to_string()))
             .guid(Some(guid))
             .pub_date(date.map(|date| date.format(&Rfc2822).unwrap()))
-            .description(description)
-            .build();
+            .description(description);
+
+        // Media enclosure
+        if let Some(img_selector) = &config.media {
+            let media = item
+                .as_node()
+                .select_first(&img_selector)
+                .map_err(|()| eyre!("invalid selector for media: {}", img_selector))?;
+
+            let media_attrs = media.attributes.borrow();
+            let media_url = media_attrs
+                .get("src")
+                .ok_or_else(|| eyre!("element selected as image has no 'src' attribute"))?;
+
+            let mut enclosure = Enclosure::default();
```

Comment:
Suggest using `EnclosureBuilder` and constructing the `Enclosure` after all the fields have been properly extracted.